Docker python api >2.0

mikaelarguedas opened the pull request:
Update cleanup_docker_images.py to match latest (docker 2.4.2) API.

Replacement of https://github.com/ros-infrastructure/buildfarm_deployment/pull/156 targeting the xenial enabled branch

Should address https://github.com/ros-infrastructure/buildfarm_deployment/issues/155

mikaelarguedas committed:
update API for python3-docker >2.0 (tested with 2.4.2)

mikaelarguedas committed:
remove now unused duplicated cleanup_docker_images.py

nuclearsandwich commented:
Deploying this to some testfarm hosts now.

mikaelarguedas commented:
:crossed_fingers:

nuclearsandwich commented:
Even with the updated API it looks like the same error still exhibits. I'm going to dig into it a little further, but this still moves us forward in a big way. Thanks @mikaelarguedas!

nuclearsandwich committed:
Use Docker Python API 2.0.

The 2.0 Docker python library removes untagged ('<none>:<none>') images
from the RepoTags image attribute so this should test for an empty list
rather then an explict none tag.
https://github.com/docker/docker-py/blob/2.4.2/docker/models/images.py#L41-L44

The remove_image function was also moved and should be accessed by the
client's `image` attribute.

nuclearsandwich committed:
Flake8 comment and line spacing cleanup.

nuclearsandwich committed:
Remove unused import.

nuclearsandwich committed:
Use up-to-date name and version for the Docker Python API package.

docker-py appears to be an abandoned copy of the docker pypi package
stuck at 1.10.6. An issue with the script on Xenial prompted us to
update the dependency.

mikaelarguedas commented on modules/agent_files/files/home/jenkins-agent/cleanup_docker_images.py:
```diff
@@ -61,7 +61,7 @@ def run_image_cleanup(args, minimum_age, dclient):
     for i in reversed(images):
         dockerid = i.id
-        repo_tags = i.attrs['RepoTags'][0] if '<none>:<none>' not in i.attrs['RepoTags'] else dockerid
+        repo_tags = i.attrs['RepoTags'][0] if i.attrs['RepoTags'] else dockerid
```
Should we access `i.tags` directly then? My understanding was that `RepoTags` had the `<none:none>` but the tags attribute had it stripped out, that's why I left it to keep the changes minimal.

mikaelarguedas commented on modules/agent_files/files/home/jenkins-agent/cleanup_docker_images.py:
```diff
@@ -76,7 +76,7 @@ def run_image_cleanup(args, minimum_age, dclient):
                     logging.info("Dry run >> I would have removed image: %s" % repo_tags)
                 else:
-                    dclient.remove_image(repo_tags)
+                    dclient.images.remove(repo_tags)
```
sorry about that :/

nuclearsandwich replied:
No worries. It was an easy fix.

nuclearsandwich commented on modules/profile/manifests/jenkins/agent.pp:
```diff
@@ -100,9 +100,9 @@
-    pkgname        => 'docker-py',
-    #ensure        => present, # defaults to present
+  python::pip { 'docker':
+    ensure  => '2.5.0',
```
A 2.5.0 release happened between the start of this adventure and this morning. It's what I got when I ran `pip3 install docker` on the test hosts so that's what I'm pinning for future hosts.

nuclearsandwich commented:
Did a bit more fixing and cleanup and it now works as desired. Thanks again for doing the work to get this rolling @mikaelarguedas
